### DISCUSSION TITLE
serverless function renders as text

### DISCUSSION THREAD

**Discussion by selfagency** [vercel/vercel#6645] [Help] [answered]
hi, i'm trying to deploy fastify to vercel, but following the [instructions on the fastify site](https://www.fastify.io/docs/latest/Serverless/), my app returns my serverless function as [a javascript text file](https://api.byte.lease/) instead of executing it.

indicating it's treating it as a static file, `vercel dev` returns:
```
[debug] [2021-08-28T19:28:25.055Z] Serving asset: [FileFsRef] api/main.js
```
 
when i try to get around this by specifying `main.js` or `api/main.js` as a `function` in the `vercel.json` configuration object, it says:
```
Error! The pattern "main.js" defined in `functions` doesn't match any Serverless Functions inside the `api` directory.
```

any idea what i may be doing wrong here? code is [here](https://github.com/selfagency/byte-lease-api). thanks!

**Comment by selfagency** [comment]
i'm actually getting really frustrated. i've been a paying vercel customer for five years and for the first time i'm considering switching providers because of the horrendous lack of support. i've been unable to deploy my project for almost a week and i'm getting absolutely nowhere. it took three days to get a response, another day to get a response back to my "i already tried that" and today i got back a response telling me to read the documentation like i haven't been an active paying customer since your platform existed with dozens of sites deployed there. first time i run into an actual problem that someone could probably resolve for me in 30 seconds if they bothered to look, and instead, i'm pulling teeth to get the most minimal help. i'm starting to feel like a sucker for proudly supporting and evangelizing for your service for all these years. @rauchg 

![CleanShot 2021-08-31 at 19 08 40](https://user-images.githubusercontent.com/2541728/131587166-d326f73c-fc89-43e3-9249-96b2a0181068.png)

**Reply by mcsdevv** [reply]
Hi @selfagency, I'm sorry you've had a poor experience here, that's the absolute last thing we want to happen. I've taken a look a look at the case you raised and I think there's some confusion on our side which has led us to interpret your answer incorrectly, either way, we could've done a much better job here, and again, I'm sorry for that.

I totally understand that on this occasion we've missed the mark and that there may not be much we can do to help with this now. If there is, or if there's anything else you feel we're not doing a good enough job of helping with in the future, please contact me [directly](mailto:matthew@vercel.com) and I'll do my best to assist as soon as possible.

**Comment by selfagency** [comment, marked as answer]
I figured out a solution but I'm not happy about my customer service experience one bit
